spec/spec_helper_acceptance.rb

```diff
@@ -98,7 +98,7 @@ def run_puppet_agent(options = { allow_changes: true })
 def run_puppet_device_generate_csr(cert_name)
   acceptable_exit_codes = 1
   on(default, puppet('device', '--verbose', '--waitforcert=0', '--target', cert_name), acceptable_exit_codes: acceptable_exit_codes) do |result|
-    expect(result.stdout).to match(%r{Exiting; no certificate found and waitforcert is disabled})
+    expect(result.stdout).to match(%r{Exiting now because the waitforcert setting is set to 0})
```

Comment:
Since we're testing against multiple versions of puppet you'll need to check for both messages.

Reply:
No good deed goes unpunished: https://tickets.puppetlabs.com/browse/PUP-3122

Reply:
We still  :yellow_heart:  you @joshcooper

Reply:
haha, yeah 👍